packages/rust-dprint-plugin-typescript/src/main.rs

```diff
@@ -0,0 +1,189 @@
+extern crate dprint_plugin_typescript as dprint;
+
+use clap::{App, Arg};
+use dprint::configuration::{Configuration, ConfigurationBuilder};
+use serde_json;
+use std::fs;
+use std::path::Path;
+use std::path::PathBuf;
+
+fn is_supported(path: &Path) -> bool {
+    if let Some(ext) = path.extension() {
+        if ext == "ts" || ext == "tsx" || ext == "js" || ext == "jsx" {
+            true
+        } else {
+            false
+        }
+    } else {
+        false
+    }
+}
+
+fn check_source_files(config: Configuration, paths: Vec<PathBuf>) -> Result<(), std::io::Error> {
+    let mut not_formatted_files = vec![];
+
+    for file_path in paths {
+        let file_path_str = file_path.to_string_lossy();
+        let file_contents = fs::read_to_string(&file_path).unwrap();
+        match dprint::format_text(&file_path_str, &file_contents, &config) {
+            Ok(None) => {
+                // nothing to format, pass
+            }
+            Ok(Some(formatted_text)) => {
+                if formatted_text != file_contents {
+                    not_formatted_files.push(file_path);
+                }
+            }
+            Err(e) => {
+                eprintln!("Error checking: {}", &file_path_str);
+                eprintln!("   {}", e);
+            }
+        }
+    }
+
+    if not_formatted_files.is_empty() {
+        Ok(())
+    } else {
+        let f = if not_formatted_files.len() == 1 {
+            "file"
+        } else {
+            "files"
+        };
+        Err(std::io::Error::new(
+            std::io::ErrorKind::Other,
+            format!("Found {} not formatted {}", not_formatted_files.len(), f,),
+        ))
+    }
+}
+
+fn format_source_files(config: Configuration, paths: Vec<PathBuf>) -> Result<(), std::io::Error> {
+    let mut not_formatted_files = vec![];
+
+    for file_path in paths {
+        let file_path_str = file_path.to_string_lossy();
+        let file_contents = fs::read_to_string(&file_path)?;
+
+        match dprint::format_text(&file_path_str, &file_contents, &config) {
+            Ok(None) => {
+                // nothing to format, pass
+            }
+            Ok(Some(formatted_text)) => {
+                if formatted_text != file_contents {
+                    println!("{}", file_path_str);
+                    fs::write(&file_path, formatted_text)?;
+                    not_formatted_files.push(file_path);
+                }
+            }
+            Err(e) => {
+                eprintln!("Error formatting: {}", &file_path_str);
+                eprintln!("   {}", e);
+            }
+        }
+    }
+
+    let f = if not_formatted_files.len() == 1 {
+        "file"
+    } else {
+        "files"
+    };
+    println!("Formatted {} {}", not_formatted_files.len(), f);
+    Ok(())
+}
+
+pub fn format(
```

Comment:
If this file is for the binary file, then should this be public?

Reply:
It shouldn't